### PR TITLE
Lazily resolve symbol table lookups for testers and updaters

### PR DESCRIPTION
**Pull request by ajreynol**
This is in preparation for the flex parser.

This corrects the ANTLR grammar and SMT state for parsing testers and updaters for overloaded selectors or constructors so that indices of these are *symbols* not *constructor terms*. To handle overloading, symbol table lookups must be done lazily.

### CODE REVIEW

File: src/parser/smt2/smt2.cpp [hunk 1]
```diff
@@ -1325,6 +1325,12 @@ Term Smt2State::applyParseOp(const ParseOp& p, std::vector<Term>& args)
     Trace("parser") << "applyParseOp: return store all " << ret << std::endl;
     return ret;
   }
+  else if (p.d_kind == APPLY_TESTER || p.d_kind == APPLY_UPDATER)
+  {
+    Term iop = mkIndexedOp(p.d_kind, {p.d_name}, args);
+    kind = p.d_kind;
+    args.insert(args.begin(), iop);
```
Comment (mudathirmahgoub) [inline src/parser/smt2/smt2.cpp:1332]: Why is the end here is a term iop? Did you mean args.end()?

Comment (ajreynol) [reply src/parser/smt2/smt2.cpp:1332]: This code is determining how to construct the operator, which should be passed as the first child to `mkTerm`. It is prepended to args.

Comment (mudathirmahgoub) [reply src/parser/smt2/smt2.cpp:1332]: Sorry, my bad. You are just inserting the term at the beginning. 
I thought this is the overloaded version of insert that copies multiple elements.

**Review by mudathirmahgoub — APPROVED** [review]
LGTM. Thanks for supporting the standard.

**Comment by ajreynol** [comment]
This is in preparation for the flex parser.

This corrects the ANTLR grammar and SMT state for parsing testers and updaters for overloaded selectors or constructors so that indices of these are symbols not constructor terms. To handle overloading, symbol table lookups must be done lazily.

**Pull request merged**